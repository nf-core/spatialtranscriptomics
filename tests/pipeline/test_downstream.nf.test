nextflow_pipeline {
    name "Test downstream workflow (excl. Space Ranger)"
    script "main.nf"
    tag "pipeline"

    test("Downstream FFPE v2 CytAssist") {
        when {
            params {
                // Input and output
                input  = 'https://raw.githubusercontent.com/nf-core/test-datasets/spatialtranscriptomics/testdata/human-brain-cancer-11-mm-capture-area-ffpe-2-standard_v2_ffpe_cytassist/samplesheet_downstream.csv'
                spaceranger_probeset = "https://raw.githubusercontent.com/nf-core/test-datasets/spatialtranscriptomics/testdata/human-brain-cancer-11-mm-capture-area-ffpe-2-standard_v2_ffpe_cytassist/outs/probe_set.csv"
                spaceranger_reference = "https://raw.githubusercontent.com/nf-core/test-datasets/spatialtranscriptomics/testdata/homo_sapiens_chr22_reference.tar.gz"

                // Parameters
                qc_min_counts = 5
                qc_min_genes = 3
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(

                // Workflow
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("nf_core_pipeline_software_mqc_versions.yml") },

                // Data
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/data/adata_processed.h5ad").exists() },
                { assert path("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/data/sdata_processed.zarr").exists() },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2.2/data/adata_processed.h5ad").exists() },
                { assert path("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2.2/data/sdata_processed.zarr").exists() },

                // Reports
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/reports/quality_controls.html").text.contains("final results of all the filtering") },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/reports/clustering.html").text.contains("spatial distribution of clusters") },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/reports/spatially_variable_genes.html").text.contains("Spatial transcriptomics data can give insight") },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2.2/reports/quality_controls.html").text.contains("final results of all the filtering") },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2.2/reports/clustering.html").text.contains("spatial distribution of clusters") },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2.2/reports/spatially_variable_genes.html").text.contains("Spatial transcriptomics data can give insight") },

                // DEGs
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2/degs/spatially_variable_genes.csv").exists() },
                { assert file("$outputDir/CytAssist_11mm_FFPE_Human_Glioblastoma_2.2/degs/spatially_variable_genes.csv").exists() },

                // MultiQC
                { assert file("$outputDir/multiqc/multiqc_report.html").exists() }
            )
        }
    }
}
