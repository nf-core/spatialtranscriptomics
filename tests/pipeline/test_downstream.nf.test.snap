{
    "nf_core_pipeline_software_mqc_versions.yml": {
        "content": [
            "{CLUSTERING={quarto=1.4.549, papermill=null}, QUALITY_CONTROLS={quarto=1.4.549, papermill=null}, READ_DATA={scanpy=1.9.8}, SPACERANGER_UNTAR_REFERENCE={untar=1.3}, SPATIALLY_VARIABLE_GENES={quarto=1.4.549, papermill=null}, UNTAR_DOWNSTREAM_INPUT={untar=1.3}, Workflow={nf-core/spatialtranscriptomics=v1.0dev}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T18:46:59.035976"
    }
}